src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,125 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (isMapLoad()) {
+            resize();
+        }
+        table = putElement(key, value, table);
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> curentNode = table[getIndex(key)];
+        while (curentNode != null) {
+            if (campareKeys(curentNode.key, key)) {
+                return curentNode.value;
+            }
+            curentNode = curentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        Node<K, V>[] oldTable = table;
+        table = new Node[table.length << 1];
+        size = 0;
+        for (int i = 0; i < oldTable.length; i++) {
+            Node<K, V> curentNode = oldTable[i];
+            while (curentNode != null) {
+                table = putElement(curentNode.key, curentNode.value, table);
+                curentNode = curentNode.next;
+            }
+        }
+    }
+
+    private Node<K, V>[] putElement(K key, V value, Node<K, V>[] table) {
+        Node<K, V> newElement = new Node<>(key, value);
+        int index = getIndex(key);
+        if (table[index] != null) {
+            if (isKeyPresent(key, table)) {
+                changeElementInNodeList(table[index], newElement);
+                return table;
+            }
+            putToNodeList(table[index], newElement);
```

Comment:
You can optimize here, instead of doing almost the same check and change, you can check and change using only one loop